pandas/core/arrays/categorical.py

```diff
@@ -1847,6 +1847,15 @@ def __iter__(self):
         """Returns an Iterator over the values of this Categorical."""
         return iter(self.get_values().tolist())
 
+    def __contains__(self, key):
+        """Returns True if `key` is in this Categorical."""
+        if key in self.categories:
+            return self.categories.get_loc(key) in self.codes
+        elif isna(key) and self.isna().any():
```

Comment:
how is ``isna`` implemented for Categoricals? I'm wondering if it'd be alot faster to instead do ``elif isna(key) and -1 in self.codes`` or if it'd be about the same is your version?

Reply:
I tested this in my #21369 and they seem both to be the same speed and the na-version reads better, so IMO thatis ok to use.